kb/migrating-visibility-data-from-es6.md

```diff
@@ -0,0 +1,82 @@
+---
+slug: migrate-visibility-data-from-es6
+title: Migrate visibility data from ES6
+tags:
+  - kb-article
+date: 2022-29-08T00:00:00Z
+---
+
+We added support for [Elasticsearch v7+ (ES7)](https://www.elastic.co/downloads/past-releases/#elasticsearch) in the v1.7.0 update to the [Temporal Server](/clusters#temporal-server).
+ES7 introduces several breaking changes, including the removal of mapping types.
+These changes make ES6 incompatible with ES7.
+
+Upgrade ES6 to ES7, and then use one of the following methods to update your Temporal Server.
+
+:::note
+
+If you are using AWS Elasticsearch, only upgrade to version 7.7.
+
+:::
+
+## Fresh installation
+
+1. Install the latest version of Temporal Server.
+2. Install ES7 as shown in the [Elasticsearch documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/setup-upgrade.html).
+3. To set the Elasticsearch version in Temporal Server:
+
+- Set the version in the server's static config file:
+
+  ```
+  persistence:
+    datastores:
+      es-visibility:
+        elasticsearch:
+          version: "v7"
+  ```
+
+- Set `ES_VERSION` to `v7` in the pre-built Docker image.
+
+## Upgrade with temporary shutdown
+
+1. Shut down the Temporal Cluster.
+2. Upgrade ES6 to ES7 according to [Elasticsearch documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/setup-upgrade.html).
+
```

Comment:
3. Upgrade Temporal to the latest version. Make sure that you don't have
 ```
  persistence:
    datastores:
      es-visibility:
        elasticsearch:
          version: "v6"
  ```
in your config file.
4. Start the Temporal Cluster.